test/game_test.exs

```diff
@@ -266,4 +270,217 @@ defmodule Gametest do
   end
+
+  test "assign_player_2_type with best should set player 2 to best" do
+    game = %{
+      :available_moves => [1, 2, 3, 4, 5, 6, 7, 8, 9],
+      :win? => false,
+      :player_1_moves => [],
+      :player_2_moves => [],
+      :player_1_mark => "X",
+      :player_2_mark => "O",
+      :player_1_type => :human,
+      :player_2_type => :human,
+      :current_player => 1
+    }
+
+    assert Game.assign_player_2_type(:best, game)[:player_2_type] == :best
+  end
+
+  test "assign_player_2_type with next should set player 2 to next" do
+    game = %{
+      :available_moves => [1, 2, 3, 4, 5, 6, 7, 8, 9],
+      :win? => false,
+      :player_1_moves => [],
+      :player_2_moves => [],
+      :player_1_mark => "X",
+      :player_2_mark => "O",
+      :player_1_type => :human,
+      :player_2_type => :next,
+      :current_player => 1
+    }
+
+    assert Game.assign_player_2_type(:next, game)[:player_2_type] == :next
+  end
+
+  test "next_open should return the next move that is available" do
+    game = %{
+      :available_moves => [1, 2, 3, 4, 5, 6, 7, 8, 9]
+    }
+
+    assert Game.next_open(game) == 1
+  end
+
+  test "handle_next_move should return the next move that is available" do
+    game = %{
+      :available_moves => [1, 2, 3, 4, 5, 6, 7, 8, 9],
+      :win? => false,
+      :player_1_moves => [],
+      :player_2_moves => [],
+      :player_1_mark => "X",
+      :player_2_mark => "O",
+      :player_1_type => :next,
+      :player_2_type => :human,
+      :current_player => 1,
+      :board => %{
+        1 => "1",
+        2 => "2",
+        3 => "3",
+        4 => "4",
+        5 => "5",
+        6 => "6",
+        7 => "7",
+        8 => "8",
+        9 => "9"
+      }
+    }
+
+    expected_game = %{
+      :available_moves => [2, 3, 4, 5, 6, 7, 8, 9],
+      :win? => false,
+      :player_1_moves => [1],
+      :player_2_moves => [],
+      :player_1_mark => "X",
+      :player_2_mark => "O",
+      :player_1_type => :next,
+      :player_2_type => :human,
+      :current_player => 1,
+      :board => %{
+        1 => "X",
+        2 => "2",
+        3 => "3",
+        4 => "4",
+        5 => "5",
+        6 => "6",
+        7 => "7",
+        8 => "8",
+        9 => "9"
+      }
+    }
+
+    assert Game.handle_next_move(game) == expected_game
+  end
+
+  test "score should return 0 when the game is a draw" do
+    game = %{
+      :available_moves => [],
+      :win? => false
+    }
+
+    assert Game.score(game) == 0
+  end
+
+  test "score should return 10 when the game is won by maximizer" do
+    game = %{
+      :win? => true,
+      :maximizer => 1,
+      :winner => 1
+    }
+
+    assert Game.score(game) == 10
+  end
+
+  test "score should return -10 when the game is won by minimizer" do
+    game = %{
+      :win? => true,
+      :maximizer => 1,
+      :winner => 2
+    }
+
+    assert Game.score(game) == -10
+  end
+
+  test "best_move returns final move that wins" do
+    game = %{
```

Comment:
this setup is getting very complex and makes your tests hard to follow

Reply:
I started changing the test names to help with the confusion. It specifies exactly what the move should be now